tests/Avalonia.Base.UnitTests/Input/KeyboardDeviceTests.cs

```diff
@@ -159,5 +159,31 @@ public void Control_Focus_Should_Be_Set_Before_FocusedElement_Raises_PropertyCha
 
             Assert.Equal(1, propertyChangedRaised);
         }
+
+        [Fact]
+        public void Cancelled_Focus_Change_Should_Not_Send_Got_Focus_Event()
```

Comment:
Can you add a unit test with focus redirection?
Is it possible by a direct focus in this PR? Or do we need something like this https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.gettingfocuseventargs.trysetnewfocusedelement?view=windows-app-sdk-1.6 ?

Reply:
Added.